azurerm/data_source_proximity_placement_group.go

```diff
@@ -0,0 +1,47 @@
+package azurerm
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmProximityPlacementGroup() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmProximityPlacementGroupRead,
+		Schema: map[string]*schema.Schema{
+			"resource_group_name": azure.SchemaResourceGroupNameForDataSource(),
+
+			"name": {
```

Comment:
Hi @katbyte,

wrt the ID it is the VM/VMSS API that is incorrect. This however is not limited to the newer PPG property. I re-used a lot of the code for adding availability sets to add PPG and I noticed that it was setting the availability set ID to lower case. I initially removed this because I thought it was odd but as it happens the availability set ID has the very same issue. We couldn't get them to fix this now because it will break a lot of people's state files.

I'll review all of the suggestions tomorrow and fix the conflicts too - a lot of which I think came out of go fmt.